bindings/java/src/main/native/jucx_common_def.h

```diff
@@ -72,24 +60,29 @@ JNIEnv* get_jni_env();
 /**
  * @brief Send callback used to invoke java callback class on completion of ucp operations.
  */
-void jucx_request_callback(void *request, ucs_status_t status);
+void jucx_request_callback(void *request, ucs_status_t status, void *user_data);
 
 /**
  * @brief Recv callback used to invoke java callback class on completion of ucp tag_recv_nb operation.
  */
-void recv_callback(void *request, ucs_status_t status, ucp_tag_recv_info_t *info);
+void recv_callback(void *request, ucs_status_t status, const ucp_tag_recv_info_t *info, void *user_data);
 
 /**
  * @brief Recv callback used to invoke java callback class on completion of ucp stream_recv_nb operation.
  */
-void stream_recv_callback(void *request, ucs_status_t status, size_t length);
+void stream_recv_callback(void *request, ucs_status_t status, size_t length, void *user_data);
 
 /**
- * @brief Utility to process request logic: if request is pointer - set callback to request context.
- * If request is status - call callback directly.
- * Returns jucx_request object, that could be monitored on completion.
+ * @ingroup JUCX_REQ
+ * @brief Utility to allocate jucx request and set appropriate java callback in it.
  */
-jobject process_request(void *request, jobject callback);
+jobject jucx_request_allocate(JNIEnv *env, jobject callback, ucp_request_param_t *param);
```

Comment:
can *param be const?

Reply:
It fills params with flags (user_data, callback, memtype).